src/Dfc.CourseDirectory.WebV2/DataStore/CosmosDb/Queries/UpdateProviderUkrlpData.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using Dfc.CourseDirectory.WebV2.DataStore.CosmosDb.Models;
+using Dfc.CourseDirectory.WebV2.Models;
+using OneOf;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.WebV2.DataStore.CosmosDb.Queries
+{
+    public class UpsertProviderUkrlpData : Provider, ICosmosDbQuery<Success>
+    {
+        public bool Update { get; set; }
```

Comment:
List properties explicitly here instead of inhering from Provider. Also better to have one Create command and a separate Update command instead a single one with parameter.

Reply:
ok make sense